src/main/kotlin/com/idan/pokemon_hub/init/PokemonInitializer.kt

```
@@ -112,7 +114,8 @@ class PokemonInitializer(private val pokemonRepository: PokemonRepository) {
         try {
             pokemonRepository.saveAll(pokemons)
             logger.info("Successfully initialized ${pokemons.size} Pokémon.")
-        } catch (e: Exception) {
+        } catch (e:  IllegalArgumentException) {
```

So why do you swallow this one and do nothing with it?

---

I figured (based in Kotlin Doc) that handling multiple exceptions should be in this form:
    try {
      // Do seomething
     // The order of catch blocks is important!
    } catch (e: IllegalArgumentException) {
        println("Something: ${e.message}")
    } catch (e: OptimisticLockingFailureException) {
        println("Something: ${e.message}")
    }
    
I was mistaken to think the first catch will propagate to the second one, my intent was to treat both with the same approach: "Failed to save Pokémon data".
Now I see that by catching the IllegalArgumentException, I actually ignore it and may potentially harm the app. My corrective action - pushed a fix to treat each catch particularly, even if it's another duplicated line of logger code. I guess I can also create a throwAndLog() helper func to treat these cases- but that may be overkill, how do you see it?

---

As long as you're not swallowing any exceptions it's better.